mailpoet/tests/integration/Cron/Workers/SendingQueue/Tasks/NewsletterTest.php

```
@@ -417,6 +417,136 @@ public function testItRemovesOpenTrackingPixelForSubscriberWithoutConsent() {
+
+  public function testItResolvesLinkShortcodesWithArgumentsForSubscriberWithoutConsent() {
+    // Link shortcodes may carry an argument ([link:action | name:value]); the
+    // stored link keeps the whole thing, so untracking has to resolve it rather
+    // than shipping the raw shortcode as an href.
+    $shortcode = '[link:mailpoet_test_custom_url | token:abc123]';
+    $resolvedUrl = 'http://example.com/resolved-custom-url';
+    $receivedArguments = null;
+    $wp = new WPFunctions();
+    $wp->addFilter(
+      'mailpoet_newsletter_shortcode_link',
+      function ($url, $newsletter, $subscriber, $queue, $arguments) use ($resolvedUrl, &$receivedArguments) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$url'. (undefined)

(UnusedFormalParameter)